netshare/utils/field.py

```diff
@@ -191,6 +206,53 @@ def getOutputDim(self) -> int:
         return 2 * self.num_bits  # type: ignore
 
 
+class Word2VecField(Field):
+    def __init__(self, word2vec_size, word2vec_cols, *args, **kwargs):
+        super(Word2VecField, self).__init__(*args, **kwargs)
+
+        self.word2vec_size = word2vec_size
+        self.norm_option = Normalization.MINUSONE_ONE
+        self.word2vec_dict_type_cols = get_word2vec_type_col(word2vec_cols)
+
+    def normalize(self, x, embed_model):
+        return np.array(
+            [get_vector(embed_model, str(xi), norm_option=True) for xi in x]
+        )
+
+    def denormalize(self, norm_x):
+        dict_annDictPair = {}
+        with open(get_annoy_type_dict_for_word2vec(), "r") as readfile:
+            dict_annDictPair = json.load(readfile)
+        dict_annoyIndex = {}
+        for type in dict_annDictPair.keys():
+            type_ann = AnnoyIndex(self.word2vec_size, "angular")
+            type_ann.load(get_annoyIndex_for_word2vec(type))
+            dict_annoyIndex[type] = type_ann
+
+        for k in self.word2vec_dict_type_cols:
+            if self.name in self.word2vec_dict_type_cols[k]:
+                word2vec_type_col = k
+                break
+        if word2vec_type_col is None:
```

Comment:
you might get here `AttributeError`. 
You probably would have found it if your write tests - but it's not mandatory, so it's your choice :)

Reply:
btw, a pretty cool feature that does what you're trying to do here is `for...else`
Google it :)